docs/content/ecs/component-generation.md

```diff
@@ -13,12 +13,28 @@ The [code generator]({{urlRoot}}/content/code-generator) uses `.schema` files to
 
 ## Overview
 
-A `struct`, which implements `Unity.Entities.IComponentData` and `Improbable.Gdk.Core.ISpatialComponentData`, is generated for each SpatialOS component. It has the same name as the corresponding schema component.
-The generation process creates a namespace for each struct according to the relevant schemalang component name: `{name of schema component}.Component`
+The code generator generates two structs for each SpatialOS component defined in schema. The generation process places these structs in namespace according to the schemalang component name: `{schema package}.{name of schema component}`. The two structs it generates are:
+
+  * A `struct` which implements `Improbable.Gdk.Core.ISpatialComponentSnapshot`. Its fully qualified name is: `{schema package}.{name of schema component}.Snapshot`.
+  * A `struct` which implements `Unity.Entities.IComponentData`, `Improbable.Gdk.Core.ISpatialComponentData`, and `Improbable.Gdk.Core.ISnapshottable<{schema package}.{name of schema component}.Snapshot>`. Its fully qualified name is: `{schema package}.{name of schema component}.Component`.
```

Comment:
Not sure whether better to write the third generic as `Improbable.Gdk.Core.ISnapshottable<Snapshot>`